### ISSUE TITLE
[10.0] PRs like this should not go into production without announcement

### ISSUE THREAD

**Issue by mohamedhagag** [odoo/odoo#24582]
@fpodoo @odony @oco-odoo @simongoffin 

https://github.com/odoo/odoo/blob/370ed2b66fbbdf78c86981616f248a5e909c53fe/addons/point_of_sale/models/pos_order.py#L222

This is for PR https://github.com/odoo/odoo/commit/5ca0ef978f350739bf2481c04ad10215d53fb362

How such behavioural change found its way into v10 and after a year of its release ??

Now after we used to have our fix to complete the entries we suddenly found that the COGS entries are duplicated and our customers complain.

troubleshooting and fixing 100s of 1000s of entries costed us time and money.

This should not go into production as per Odoo rules and should go in master only.

There must be an announcement before releasing into production .

this PR fixes https://github.com/odoo/odoo/issues/2049

**Comment by pedrobaeza** [comment]
@mohamedhagag several advices from a community member like you:

* Pinging everyone and more people that is not directly involved in the issue gets the contrary effect that you want, that is getting attention to this.
* The tone also doesn't help. Things are done because a reason, and this one should be an strong one. Instead of starting a sentence "How such change...", try to see what is the reason and what conflicts with your use case, and explain this.
* Not giving full details of the steps to reproduce, the exact problem, etc, makes this more difficult to be solved, and more without enterprise support that is on voluntary base.

**Comment by mohamedhagag** [comment]
@pedrobaeza Appreciate your kind advices, I wrote on hurry but I will take care next time .
regarding the detailed info about the issue there are more info in the PR and 3 years old Issue mentioned.

**Comment by gustavovalverde** [comment]
I also agree that this kind of changes should be announced in the mailing list, it's a major change in the behavior of the accounting when using the Point of Sale.

I've made simple PRs with a one-liner that have been rejected because of the stable policy; if there's an exception like this one, people should be notified.

**Comment by Yenthe666** [comment]
@mohamedhagag,

Honestly, I agree with the point that this should maybe communicated better and that this may come as a surprise. We tend to notice issues like this too but this is discussable. It is hard for Odoo to notify everybody in a proper way and about the right things. I honestly always check PR's (atleast the title) to have a global idea of what happens and moves.
If I look at the PR strictly speaking there is no breach of the stable policy though. There are no new fields or models changed and there is no introduction of new functionalities.

P.S: This is my personal opinion, this is not an official statement and I do not work at Odoo. My five cents.

**Comment by pedrobaeza** [comment]
@mohamedhagag again I think the tone you get in your answers is not the proper one for having this solved.

I still don't understand where is the problem any way. I'm understanding that you solve the problem another way out of the official ways, and now you get duplicated entries to that fix you made?

**Comment by mohamedhagag** [comment]
Dears

I'm not asking for a revert of the PR or a solution to the resulting problem, what happened is done anyway, but if we care about the community and the users we should make an announcement to save the ppl some effort and time, as every day a mid-sized POS entity can make 100s of orders and this will generate lots of duplicated accounting entries that will be hard to identify and remove or fix.

Just official announcement that will least support us in front of our customers at is more than enough.

Regards,

**Comment by pedrobaeza** [comment]
@mohamedhagag the official communication is GitHub itself, seeing new incoming commits, that are extensively documented, and checking if you want to `git pull` or not. Don't overhead things if you want to grow and evolve the way Odoo is doing.

**Comment by mohamedhagag** [comment]
@pedrobaeza 
Regarding the tone, I really can't see what's the problem with it, I'm trying to choose calm words as I can, maybe it's my weak English.

Regarding the problem, the story started more than 3 yrs ago as per issue #2049 that POS doesn't create COGS entries and this issue unresolved till the mentioned PR appeared.

The resulting issue is that every POS user had his own fix for the issue to complete the missing COGS entries and working like that for 2 years at least.

Now this PR introduced the fix which will duplicate the entries, based on this we got issues and complains from our customers that P&L showing huge loss instead of profit.

We managed to fix the duplicated entries and remove our own fix for the old issue, but this costed us some time and effort that we could avoid if we knew this before its release.

Finally this issue have been opened to warn the users and to tell the developers to take care next time while merging PRs like the one mentioned.

We are all Odoo and supporting their efforts, we helped Odoo and will continue to help and support by marketing by development and by reporting issues .

Thanks and best wishes for @simongoffin @fpodoo @pedrobaeza @Yenthe666 and all Odoo S.A. staff and the great Odoo community.

Regards,

**Comment by pedrobaeza** [comment]
Thanks for clarifying that this is not an Odoo problem, but a conflict with your own fix. I have to say that I have been affected personally by this in some occasions, and yes, it's not the ideal for customers, but it's detected, fixed removing my own patch, and going on, and what we have to be happy is to have this on core instead of own custom patches. 

I know that this is not satisfying for you at you still have the problem in this case, but let's try all to avoid this to happen again (having so much time an opened issue). Any way, you know that having an enterprise contract allows you to put a ticket for this to be solved ASAP. The rest are based on a best effort initiative, so it's not mandatory to have it.

Closing this issue and thanks for the comprehension.
